pkg/cmd/roachtest/tests/jobs.go

```diff
@@ -103,31 +102,39 @@ func jobSurvivesNodeShutdown(
 			return errors.New("job never created")
 		}
 
-		// Shutdown a node after a bit, and keep it shutdown for the remainder
-		// of the job.
-		timeToWait := 5 * time.Second
-		timer := timeutil.Timer{}
-		timer.Reset(timeToWait)
-		select {
-		case <-ctx.Done():
-			return errors.Wrapf(ctx.Err(), "stopping test, did not shutdown node")
-		case <-timer.C:
-			timer.Read = true
-		}
-
-		// Sanity check that the job is still running.
+		// Check once a second to see if the job has started running.
 		watcherDB := c.Conn(ctx, watcherNode)
 		defer watcherDB.Close()
-
-		var status string
-		err := watcherDB.QueryRowContext(ctx, `SELECT status FROM [SHOW JOBS] WHERE job_id=$1`, jobID).Scan(&status)
-		if err != nil {
-			return errors.Wrap(err, "getting the job status")
-		}
-		jobStatus := jobs.Status(status)
-		if jobStatus != jobs.StatusRunning {
-			return errors.Newf("job too fast! job got to state %s before the target node could be shutdown",
-				status)
+		timeToWait := time.Second
+		timer := timeutil.Timer{}
+		jobRunning := false
+		for {
+			var status string
+			err := watcherDB.QueryRowContext(ctx, `SELECT status FROM [SHOW JOBS] WHERE job_id=$1`, jobID).Scan(&status)
+			if err != nil {
+				return errors.Wrap(err, "getting the job status")
+			}
+			switch jobs.Status(status) {
+			case jobs.StatusPending:
+			case jobs.StatusRunning:
+				jobRunning = true
+			default:
+				return errors.Newf("job too fast! job got to state %s before the target node could be shutdown",
+					status)
+			}
+			t.L().Printf(`status %s`, status)
+			timer.Reset(timeToWait)
+			select {
+			case <-ctx.Done():
+				return errors.Wrapf(ctx.Err(), "stopping test, did not shutdown node")
+			case <-timer.C:
+				timer.Read = true
+			}
+			// Break a second after confirming the job is running to ensure the node shutdown
+			// "in the middle" of running the job, right after the job began running.
+			if jobRunning == true {
+				break
+			}
```

Comment:
I think in our original discussion we talked about how cancelling in the middle of the job might be a side-effect of the old code, which is why it is preserved here.  Perhaps lets revisit this if we are still having issues after this fix is in since we can probably come up with either better coordination or just immediately try to cancel once we know it is running.